adapters/repos/db/inverted/bm25_searcher.go

```diff
@@ -223,18 +223,29 @@ func (b *BM25Searcher) BM25F(ctx context.Context, className schema.ClassName, li
 	filter *filters.LocalFilter, sort []filters.Sort, additional additional.Properties,
 	objectByIndexID func(index uint64) *storobj.Object,
 ) ([]*storobj.Object, []float32, error) {
-	terms := helpers.TokenizeText(keywordRanking.Query)
+	textTerms := helpers.TokenizeText(keywordRanking.Query)
+	stringTerms := helpers.TokenizeString(keywordRanking.Query)
 
-	idLists := make([]docPointersWithScore, len(terms))
+	idLists := []docPointersWithScore{}
 
-	for i, term := range terms {
+	for _, term := range textTerms {
 
-		ids, err := b.retrieveForSingleTermMultipleProps(ctx, className, objectByIndexID, keywordRanking.Properties, term, keywordRanking.Query)
+		ids, err := b.retrieveForSingleTermMultipleProps(ctx, className, objectByIndexID, keywordRanking.Properties, term, "", keywordRanking.Query)
 		if err != nil {
 			return nil, nil, err
 		}
 
-		idLists[i] = ids
+		idLists = append(idLists, ids)
+	}
+
+	for _, term := range stringTerms {
+
+		ids, err := b.retrieveForSingleTermMultipleProps(ctx, className, objectByIndexID, keywordRanking.Properties, "", term, keywordRanking.Query)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		idLists = append(idLists, ids)
```

Comment:
Instead of appending you could do 
```
idLists := make([]docPointersWithScore, len(textTerms) + len(stringTerms))
for i, term := range textTerms {
    ....
    idLists[i] = ids
}
for i, term := range textTerms {
    ....
    idLists[len(textTerms) + i] = ids
}

```


Otherwise you have a lot of temporary allocations.

Reply:
There aren't more than 65k properties according to your patch, but I can add this in :)